libs/components/colorpicker/testing/src/colorpicker/colorpicker-dropdown-harness.ts

```diff
@@ -0,0 +1,169 @@
+import { HarnessPredicate, TestElement } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyColorpickerDropdownHarnessFilters } from './colorpicker-dropdown-harness.filters';
+
+/**
+ * Harness for interacting with colorpicker dropdown in tests.
+ */
+export class SkyColorpickerDropdownHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = '.sky-colorpicker-container';
+
+  #getApplyButton = this.locatorFor('.sky-btn-colorpicker-apply');
+  #getCancelButton = this.locatorFor('.sky-btn-colorpicker-close');
+  #getInputs = this.locatorForAll('input.sky-form-control');
+  #getSwatchesDefault = this.locatorForAll('button.sky-preset-color');
+  #getSwatchesModern = this.locatorForAll('button.sky-btn-link');
+
+  public static with(
+    filters: SkyColorpickerDropdownHarnessFilters,
+  ): HarnessPredicate<SkyColorpickerDropdownHarness> {
+    return new HarnessPredicate(SkyColorpickerDropdownHarness, filters);
+  }
+
+  /**
+   * Whether transparency is allowed.
+   */
+  public async allowsTransparency(): Promise<boolean> {
+    return !!(await this.#getAlphaInput());
+  }
+
+  /**
+   * Clicks the colorpicker dropdown apply button.
+   */
+  public async clickApplyButton(): Promise<void> {
+    return (await this.#getApplyButton()).click();
+  }
+
+  /**
+   * Clicks the colorpicker dropdown cancel button.
+   */
+  public async clickCancelButton(): Promise<void> {
+    return (await this.#getCancelButton()).click();
+  }
+
+  /**
+   * Clicks a specified swatch in the color preset section.
+   * @param swatchHex Hex code of the swatch to click.
+   */
+  public async clickPresetColorSwatch(swatchHex: string): Promise<void> {
+    const swatches = await this.#getSwatchButtons();
+    const ariaLabel = `Preset Color: ${swatchHex}`;
+
+    for (const swatch of swatches) {
+      if ((await swatch.getAttribute('aria-label')) === ariaLabel) {
+        return await swatch.click();
+      }
+    }
+  }
+
+  /**
+   * Gets an array of the hex codes of the preset swatches.
+   */
+  public async getPresetSwatches(): Promise<(string | undefined)[]> {
```

Comment:
Small, but we have `clickPresetColorSwatch` above; could we call this method `getPresetColorSwatches`?

Reply:
Thanks for fixing the return type of this method. The first question still applies, however.